geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala

```diff
@@ -0,0 +1,83 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.kafka.consumer
+
+import java.io.Closeable
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.{Executors, TimeUnit}
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.kafka.clients.consumer.{Consumer, ConsumerRecord}
+import org.apache.kafka.common.errors.WakeupException
+
+import scala.util.control.NonFatal
+
+trait ThreadedConsumer extends Closeable with LazyLogging {
+
+  protected val consumers: Seq[Consumer[Array[Byte], Array[Byte]]]
+  protected val topic: String
+  protected val frequency: Long
+
+  protected def consume(record: ConsumerRecord[Array[Byte], Array[Byte]]): Unit
+
+  private val closed = new AtomicBoolean(false)
+
+  private val executor = Executors.newFixedThreadPool(consumers.length)
+
+  consumers.foreach(c => executor.execute(new ConsumerRunnable(c, frequency)))
+
+  logger.debug(s"Started ${consumers.length} consumer(s) on topic $topic")
+
+  override def close(): Unit = {
+    closed.set(true)
+    consumers.foreach(_.wakeup())
+    executor.shutdownNow()
+    executor.awaitTermination(1, TimeUnit.SECONDS)
+  }
+
+  class ConsumerRunnable(val consumer: Consumer[Array[Byte], Array[Byte]], frequency: Long) extends Runnable {
+
+    private var errorCount = 0
+
+    override def run(): Unit = {
+      try {
+        while (!closed.get()) {
+          try {
+            val result = consumer.poll(frequency)
+            logger.debug(s"Consumer poll received ${result.count()} records from topic $topic")
```

Comment:
may be nice to have an ID here of some sort so you know which consumer is logging but not a big deal.

Reply:
hmm, i don't see any identifiers for consumers - i could add a number to each thread?

Reply:
yeah that's what i was thinking...we have that problem elsewhere...you have to override the threadfactory most likely. kinda gross.

Reply:
or use guava :)

Reply:
haha, i could just add a val to the `ConsumerRunnable` when i create it - i'll go ahead and do that

Reply:
sounds good.